File: Model/Event.cs
using System;
using System.Text.Json.Serialization;

namespace mpp_proiect_csharp_DianaGliga11.Model
{
    [Serializable]
    public class Event : Entity<long>
    {
        [JsonPropertyName("distance")]
        public int Distance { get; set; }
        
        [JsonIgnore] // opțional, ca să nu fie serializată
        public string DisplayText => $"{Style} - {Distance}m";


        [JsonPropertyName("style")]
        public String Style { get; set; }
        //private List<Participant>  participants;

        [JsonConstructor]
        public Event()
        {
        }

        public Event( String style, int distance)
        {
            Style = style;
            Distance = distance;
            //this.participants = participants;
        }

        public override String ToString()
        {
            return $"ID={Id}, Distance={Distance}, style={Style}";
        }

        public override bool Equals(object obj)
        {
            if (obj is Event other)
                return this.Id == other.Id;
            return false;
        }

        public override int GetHashCode()
        {
            return Id.GetHashCode();
        }

    }
}

File: flutter_client/lib/models/event.dart
// class Event {
//   final int id;
//   final String style;
//   final int distance;
//
//   Event({required this.id, required this.style, required this.distance});
//
//   factory Event.fromJson(Map<String, dynamic> json) {
//     return Event(
//       id: json['id'],
//       style: json['style'],
//       distance: json['distance'],
//     );
//   }
//
//   Map<String, dynamic> toJson() {
//     return {
//       'id': id,
//       'style': style,
//       'distance': distance,
//     };
//   }
// }

class Event{
  Event({
    required this.id,
    required this.style,
    required this.distance
});

  final int id;
  final String style;
  final int distance;
}